lib/text/lzw.flow

```diff
@@ -0,0 +1,73 @@
+import string;
+import ds/list;
+import ds/tree;
+import text/blueprint;
+
+export {
+	// LZW-compress a string
+	lzw_encode(s : string) -> [int];
+
+	// Decompress an LZW-encoded string
+	lzw_decode(data : [int]) -> string;
+}
+
+lzw_encode(s : string) -> [int] {
+	dict : ref Tree<[int], int> = ref makeTree();
+	out : ref List<int> = ref makeList();
+	phrase : ref [int]= ref [getCharCodeAt(s, 0)];
+	code = ref 65536;
+	fori(1, strlen(s)-1, \i -> {
```

Comment:
use foldi to keep it in functional style with some nice struct inside

Reply:
I'm afraid it will increase the load on the GC :-(

Reply:
OK, probably you are right